client/src/components/SinglePost.js

```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import { useState } from "react";
+import { Link } from "react-router-dom";
+// import "../App.css"
+
+//brings you to single post
+//fetches data for that individual post
+export default function SinglePost(props) {
+  let postID = props.match.params.singlePostID;
+  const [postInfo, setPostInfo] = useState(null);
+
+  fetch(`/view/${postID}`)
+    .then((res) => res.json())
+    .then((res) => {
+      setPostInfo(res);
+    });
+
+  //return brings form to edit.... cannot get to function, i believe form action and method is issue
+  return (
+    <div>
+      {postInfo === null ? null : (
+        <div>
+         <h1 id="edit title">Edit: </h1>
+          <form id="editform"action={`/Edit/${postID}`} method="POST">
+            <input id="edit-title"name="title" placeholder={postInfo.title} type="text" />
+
+            <input id="edit-author"name="author" placeholder={postInfo.author} type="text" />
+
+            <textarea id="edit-text"
+              name="content"
+              placeholder={postInfo.content}
+              type="text"
+            />
```

Comment:
Using the placeholder here is why your text isnt fully filled in here and when you submit you get empty fields. You can do this differntly by using `value={postInfo.author}` and having an `onChange={aFunction}` that would update state you can maintain all of your other values when you submit it